src/topotoolbox/gridmixins/fillsinks.py

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+import ctypes
+import os
+import copy
+
+class FillsinksMixin():
+    def fillsinks(self, maxdepth=None, sinks=None):
+
+        #TODO: arguments need to be evaluated.
+
+        basedir = os.path.dirname(__file__)
+        parentdir = os.path.dirname(basedir)
+        libpath = os.path.join(parentdir, 'libtopotoolbox.so')
+
+        dll = ctypes.CDLL(libpath)
+
+        dll.fillsinks.argtypes = [
+            np.ctypeslib.ndpointer(dtype=np.float32, flags='C_CONTIGUOUS'), # output
+            np.ctypeslib.ndpointer(dtype=np.float32, flags='C_CONTIGUOUS'), # dem
```

Comment:
Using Fortran order instead of C order should work seamlessly with all NumPy and Matplotlib functions. So, I don't think it's even necessary to convert it back.
```python
self.z = np.asfortranarray(dataset.read(1), dtype=np.float32)
```
should work instead of
```python
self.z = dataset.read(1).astype(np.float32)
```